iothub_client/src/iothub_client_ll_uploadtoblob.c

```diff
@@ -66,6 +67,10 @@ typedef struct IOTHUB_CLIENT_LL_UPLOADTOBLOB_HANDLE_DATA_TAG
         UPLOADTOBLOB_X509_CREDENTIALS x509credentials; /*assumed to be used when both deviceKey and deviceSasToken are NULL*/
     } credentials;                              /*needed for file upload*/
     char* certificates; /*if there are any certificates used*/
+	char* http_proxy_hostname;
+	int http_proxy_port;
+	char* http_proxy_username;
+	char* http_proxy_password;
```

Comment:
Would you want to add an HTTP_PROXY_OPTION object here instead of defining each variable?  You would have to do some casting from const char* to char*, but it may make it easier.